stories-table-add-search

From: Bryan Larsen <bryan@larsen.st>

To get the search feature working, we need to update the controller side. Add a `show` method to `app/controllers/projects_controller.rb` and update the `<table-plus>` to use `@stories`:

SHOW_PATCH

(To do -- explain how that works!)
---
 app/controllers/projects_controller.rb |    7 +++++++
 app/views/projects/show.dryml          |    2 +-
 2 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/app/controllers/projects_controller.rb b/app/controllers/projects_controller.rb
index 210b07d..a14339c 100644
--- a/app/controllers/projects_controller.rb
+++ b/app/controllers/projects_controller.rb
@@ -4,4 +4,11 @@ class ProjectsController < ApplicationController
 
   auto_actions :all
 
+  def show
+    @project = find_instance
+    @stories =
+      @project.stories.apply_scopes(:search    => [params[:search], :title],
+                                    :order_by  => parse_sort_param(:title, :status))
+  end
+
 end
diff --git a/app/views/projects/show.dryml b/app/views/projects/show.dryml
index 9e1fa90..463da0c 100644
--- a/app/views/projects/show.dryml
+++ b/app/views/projects/show.dryml
@@ -1,6 +1,6 @@
 <show-page>
   <collection: replace>
-  <table-plus:stories fields="this, tasks.count, status">
+  <table-plus with="&@stories" fields="this, tasks.count, status">
     <empty-message:>No stories match your criteria</empty-message:>
   </table-plus>
   </collection:>
